django-rgd/client/rgd_client/plugin.py

```diff
@@ -118,3 +119,33 @@ def create_file_from_url(
 
         r.raise_for_status()
         return r.json()
+
+    def download_file(self, id: int, download_location: Optional[str] = None) -> Path:
+        """
+        Download a ChecksumFile.
+
+        Args:
+            id: The id of the ChecksumFile
+            download_location: Path to download file to. Defaults to current working directory.
+
+        Returns:
+            The path of the newly downloaded file
+        """
+        r = self.session.get(f'checksum_file/{id}/data')
+        r.raise_for_status()
+
+        if download_location is not None:
+            download_location_path = Path(download_location)
+        else:
+            # Download to current working directory if download_location isn't given
+            download_location_path = Path.cwd()
+
+        # Create the download directory if it doesn't exist
+        download_location_path.mkdir(parents=True, exist_ok=True)
+
+        # Save the file
+        with open(download_location_path / str(id), 'wb') as file:
```

Comment:
We should probably attempt to use the ChecksumFile name here, defaulting to the ID if there's no name for that file (or rather, if the name is the empty string, as I don't think that field is nullable).

Reply:
they are unique per fileset / collection. Could you also implement a `download_file_set` method here too?

Reply:
> I agree that would be much nicer, but ChecksumFile names aren't unique.

I still think it'd be okay to download with the name, even if they aren't unique, since this method if just for a single file. If someone is individually downloading multiple files with the same name to the same path, imo that's kind of on them. We could also add something like a `use_id` argument (defaulting to False), if someone really wanted to avoid the file names.

Reply:
+1^

Reply:
> We should probably attempt to use the ChecksumFile name here, defaulting to the ID if there's no name for that file (or rather, if the name is the empty string, as I don't think that field is nullable).

When I read this initially I didn't look close enough at the code snippet and thought you were saying allow users to query for a ChecksumFile by its name; rereading this I'm realizing that you were referring to the name of the downloaded file. That makes much more sense :)